### PR TITLE
Fix for TableFromJsonArray: handling empty arrays with specified TableType #2267

### PR DESCRIPTION
**Pull request by faramos**
Fix to #2267 with failure test cases

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Json/FormulaValueJSON.cs [hunk 1]
```diff
@@ -208,7 +208,8 @@ internal static FormulaValue TableFromJsonArray(JsonElement array, TableType tab
 
             // Single Column table (e.g. [1,2,3]) Pattern for table is unique
             // since in that case nested elements are not object and hence needs to be handled differently.
-            var nestedElementsAreObjects = array.EnumerateArray().Any(nestedElement => nestedElement.ValueKind == JsonValueKind.Object);
+            var nestedElementsAreObjects = (tableType != null && tableType.FieldNames.Any(name => name != "Value"))
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Json/FormulaValueJSON.cs:211]: This is the key change

**Review by MikeStall — DISMISSED** [review]

File: src/libraries/Microsoft.PowerFx.Core/Binding/BindInfo/NameLookupInfo.cs [hunk 2]
```diff
@@ -47,22 +47,21 @@ public NameLookupInfo(BindKind kind, DType type, DPath path, int upCount, object
...
-            {
-                Name = ns.Name,
-                DisplayName = this.DisplayName,                
-                Properties = ns.Props,
-                Type = FormulaType.Build(this.Type),
-                Slot = ns
-            };
-            return true;
+            x = null;
+            return false;
```
Comment (LucGenetier) [inline src/libraries/Microsoft.PowerFx.Core/Binding/BindInfo/NameLookupInfo.cs:64]: Even if this is cleaner, I wouldn't make any code change here

File: src/libraries/Microsoft.PowerFx.Core/Public/Config/SlotMap.cs [hunk 3]
```diff
@@ -49,7 +49,7 @@ public int Alloc()
             _lowest = _items.Count + 1;
 
             _count++;
-            _items.Add(default(T));
+            _items.Add(default);
```
Comment (LucGenetier) [inline src/libraries/Microsoft.PowerFx.Core/Public/Config/SlotMap.cs:52]: >_items.Add(default); [](http://example.com/codeflow?start=12&length=20)

Same here, this is not required

File: src/tests/Microsoft.PowerFx.Json.Tests/ParseJSONTests.cs [hunk 4]
```diff
@@ -547,6 +548,39 @@ public void FromJsonHeterogeneousPrimitiveArray(string json, string expectedType
...
+        [Theory]
+        [InlineData("[{\"Name\": \"Peter\"}]", "[]")]
+        [InlineData("[1,2,3]", "[]")]
+        [InlineData("[1,2,3]", "[5, 6]")]
+        [InlineData("[\"A\", \"B\"]", "[\"1\"]")]
+        [InlineData("[{\"Name\": \"Peter\"}]", "[{\"Name\": \"Julia\"}]")]
+        [InlineData("[]", "[]")]
+        public void FromArrayWithType(string prototypeJson, string valueJson)
+        {
+            FormulaValue prototypeValue = FormulaValueJSON.FromJson(prototypeJson);
```
Comment (LucGenetier) [inline src/tests/Microsoft.PowerFx.Json.Tests/ParseJSONTests.cs:562]: >FormulaValue prototypeValue = FormulaValueJSON.FromJson(prototypeJson); [](http://example.com/codeflow?start=12&length=71)

You could add an extra test here

FormulaValue prototypeValue2 = FormulaValueJSON.FromJson(prototypeJson, prototypeValue.Type); // with specified type

and verify 

Assert.Equal(prototypeValue.Type, prototypeValue2.Type);

**Review by LucGenetier — CHANGES_REQUESTED** [review]
:clock1:

**Pull request closed without merging**